Core Tweak for reciving v2 objects

Achoobert opened the pull request:
I'm trying to track the tweaks I made to get the sdc app working with staging2, it would probibally be better to switch all of this to V2

Achoobert committed:
[fix] normalizeData, check for data.data and return

jamesduncan reviewed (changes requested):
Would you want to move this up one line so the comment about converting to an array is above the conversion? Also go ahead and add a comment so future developers can know why the data might be found in data.data.

Achoobert committed:
[fix] populate can be an array of fields

Don't overwrite the field if it exists

Achoobert committed:
[fix] access the correct directory

we are keeping ABQL in `qlOld`

Achoobert committed:
[fix] check `populate` is a valid value

I had a logic error

Achoobert committed:
[add] populate to default value

This code is needed for array to make it from definitions up to the app

Achoobert committed:
[fix] default populate is true

Achoobert committed:
[fix] prevent populate logic error

Should overwrite *only* if populate Is *neither* an array of field names, or a boolean
